src/test/java/org/cactoos/http/HtHeadTest.java

```diff
@@ -106,4 +108,52 @@ public void largeText() throws IOException {
         );
     }
 
+    @Test
+    public void edgeOfTheBlockTearing() throws IOException {
+        final int size = 16384;
+        final String header = new JoinedText(
+            "\r\n",
+            "HTTP/1.1 200 OK",
+            "Referer: http://en.wikipedia.org/wiki/Main_Page#\0",
+            "Content-type: text/plain",
+            ""
+        ).asString();
+        final String block = header.replace(
+            "\0",
+            new JoinedText(
+                "",
+                new RepeatedText(
+                    "x",
+                    size - header.length() + 1
+                ).asString()
+            ).asString()
+        );
+        MatcherAssert.assertThat(
+            block.length(),
+            new IsEqual<>(
+                size
+            )
+        );
+        MatcherAssert.assertThat(
```

Comment:
@ilyakharlamov use cactoos-matchers `Assertion` instead of `assertThat`

Reply:
@victornoel 
As far as I understand Assertion is coming in cactoos-matchers 0.13 which has not been released yet, so it's not available in maven, so can't use it yet.